types/src/database_sql.rs

```diff
@@ -226,6 +225,83 @@ impl Database {
         }))
     }
 
+    pub async fn upsert_collection(
+        &self,
+        title: String,
+        wordpress_menu_id: i64,
+        slug: String) -> Result<String>{
+        query_file!(
+            "queries/upsert_collection.sql",
+            slug,
+            title,
+            wordpress_menu_id)
+            .execute(&self.client)
+            .await?;
+        Ok(slug)
+    }
+
+    pub async fn insert_all_chapters(
+        &self,
+        chapters: Vec<Chapter>,
+        slug: String) -> Result<String>{
+
+        let mut tx = self.client.begin().await?;
+
+        // Delete all chapter data before re-inserting
+        query_file!(
+            "queries/delete_chapters.sql")
+        .execute(&mut tx)
+        .await?;
+        
+        let mut chapter_stack = Vec::new();  
+        let initial_tuple = (0, slug.clone());
+        chapter_stack.push(initial_tuple);
+
+        for current_chapter in chapters {
+
+            // Get chapter Doc ID from document title --> do in query
+            let chapter_doc_name = current_chapter.document_title;
```

Comment:
The document title does get used but the comment is a bit confusing, making it seem as if `chapter_doc_name` is supposed to be the Doc ID. The query itself gets id from the title like the comment says, but since that happens in the query it seems odd to say that here. You can just say we need the document title, not the document ID.